### ISSUE TITLE
Error "Resource not accessible by integration"

### ISSUE THREAD

**Issue by RunningLeon** [actions/stale#948]
**Description:**
Stale workflow failed when closing issues.
```
 Error: [#1728] Error when creating a comment: Resource not accessible by integration
  Error: [#1728] Error when adding a label: Resource not accessible by integration
```
log url: https://github.com/open-mmlab/mmdeploy/actions/runs/4301174276/jobs/7498093123#step:2:522
**Action version:**
stale v7

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Repro steps:**  
A description with steps to reproduce the issue. If your have a public example or repo to share, please provide the link.

**Expected behavior:**
A description of what you expected to happen.

**Actual behavior:**
A description of what is actually happening.

**Comment by dsame** [comment]
Hello @RunningLeon , thanks for your input we are starting to review the reported problem

**Comment by dsame** [comment]
Hello @RunningLeon 

can you please confirm you do not have custom github token for the repository? 

From the [log](https://github.com/open-mmlab/mmdeploy/actions/runs/4301174276/jobs/7498093123#step:1:16) is the token does not have `issues: write` permission

```
GITHUB_TOKEN Permissions
  Contents: read
  Issues: read
  Metadata: read
  PullRequests: read
```

despite the proper permissions requested in the [workflow file](https://github.com/open-mmlab/mmdeploy/actions/runs/4301174276/workflow#L12):

```
permissions:
  contents: write # only for delete-branch option
  issues: write
  pull-requests: write
```

**Comment by RunningLeon** [comment]
@dsame Hi, thanks. But the github_token is set to have read/write permission in action settings
![image](https://user-images.githubusercontent.com/28671653/222661629-1daa8b32-78e1-46a1-9a9e-647026802f6d.png)

**Comment by dsame** [comment]
Hello @RunningLeon, can you please double check the enterprise to which the repo belongs to 

![image](https://user-images.githubusercontent.com/245540/223459185-ee2679f5-97e6-44ab-8625-2a9fc0a02322.png)

does not have restricted polices:

 - In the enterprise account sidebar, click  Policies.
 - Under " Policies", click Actions.
 - Under "Workflow permissions", choose whether you want the GITHUB_TOKEN to have read and write access for all scopes, or just read access

**Comment by dsame** [comment]
Hello @RunningLeon , this is a gentle remainder the issue is staled waiting for response and to be closed soon due to inactivity.

**Comment by dsame** [comment]
Hello @RunningLeon , i am closing the issue because of long inactivity, but please feel free to reopen it or create a new one in case problem still exists

**Comment by RunningLeon** [comment]
@dsame Hi, it appeared that once the pr is merged to the master branch, the permission for `GITHUB_TOKEN` works fine(with write permission for issues as set in workflow yaml). At PR stage, the permission is not right. But this github procedure is not right for some cases. For instance, we add a workflow that requires GITHUB_TOKEN to have write permission in a PR, but the workflow cannot run due to he permission at PR stage. So we don't know whether the code is good to merge to master branch.

**Comment by dsame** [comment]
Hello @RunningLeon , thanks for the additional info - it is really helpful and from it i see it is much better ask this question in [runner](https://github.com/actions/runner) repo - its team is responsible for the described malfunction.

Do you want me to create the new issue there in behalf of you or you'd prefer to create it yourself?

**Comment by RunningLeon** [comment]
Hi, thanks for your reply. Maybe I could create a new issue later.

**Comment by dsame** [comment]
The issue is closed due it relates to other repo.

**Comment by ThulinaWickramasinghe** [comment]
I'm getting a similar issue. Following are the logs I get.

with:
    stale-pr-message: This PR has been marked stale because it has been open for 21 days with no activity.
    close-pr-message: This PR has been automatically closed since it has been in stale state for 7 days.
    days-before-stale: 21
    days-before-close: 7
    repo-token: ***
    stale-issue-label: Stale
    close-issue-reason: not_planned
    stale-pr-label: Stale
    exempt-all-milestones: false
    operations-per-run: [3](https://github.com/clubpay/customer-web-app/actions/runs/11423954871/job/31783719035#step:2:3)0
    remove-stale-when-updated: true
    debug-only: false
    ascending: false
    delete-branch: false
    exempt-all-assignees: false
    exempt-draft-pr: false
    enable-statistics: true
    ignore-updates: false
    include-only-assigned: false
The saved state was not found, the process starts from the first issue.
Starting the stale action process...
Error: Error: Getting issues was blocked by the error: Resource not accessible by integration
Error: Getting issues was blocked by the error: Resource not accessible by integration
